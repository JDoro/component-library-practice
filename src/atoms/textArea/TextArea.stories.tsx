import type {Meta, StoryObj} from '@storybook/react';

import {TextArea} from './TextArea.component.tsx';
import {useArgs} from '@storybook/preview-api';
import {fn} from '@storybook/test';

const meta = {
  title: 'Atoms/Text Area',
  component: TextArea,
  tags: ['autodocs'],
  args: {
    label: 'Enter text here',
    onChange: fn(),
  },
  render: (args) => {
    const [{ value, onChange }, updateArgs] = useArgs();
    return (
      <TextArea
        {...args}
        value={value}
        onChange={(value) => {
          onChange?.(value);
          updateArgs({ value });
        }}
      />
    );
  }
} satisfies Meta<typeof TextArea>;

export default meta;
type Story = StoryObj<typeof meta>;

export const All: Story = {};

