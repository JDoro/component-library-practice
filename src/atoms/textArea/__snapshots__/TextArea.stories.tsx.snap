// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Text Area All smoke-test 2`] = `
<div class="flex flex-col">
  <label for="textarea"
         class="text-md mb-1"
  >
    Enter text here
  </label>
  <textarea id="textarea"
            class="border-[1px] border-secondary rounded-md"
            style
  >
  </textarea>
</div>
`;
